Services/Simulation/JavascriptInterpreter.cs

```diff
@@ -178,26 +192,34 @@ private void UpdateState(JsValue data)
         {
             string key;
             object value;
-            Dictionary<string, object> stateChanges;
+            Dictionary<string, object> stateChanges = this.JsValueToDictionary(data);
 
             this.log.Debug("Updating state from the script", () => new { data, this.deviceState });
 
-            stateChanges = this.JsValueToDictionary((JsValue) data);
-
-            //Update device state with the script data passed
+            // Update device state with the script data passed
             lock (this.deviceState)
             {
                 for (int i = 0; i < stateChanges.Count; i++)
                 {
                     key = stateChanges.Keys.ElementAt(i);
                     value = stateChanges.Values.ElementAt(i);
-                    if (this.deviceState.ContainsKey(key))
-                    {
-                        this.log.Debug("state change", () => new { key, value });
-                        this.deviceState[key] = value;
-                    }
+                    this.log.Debug("state change", () => new { key, value });
+                    this.deviceState.Set(key, value);
                 }
             }
         }
+
+        // TODO: Move this out of the scriptinterpreter class into DeviceStateActor to keep this class stateless
```

Comment:
Can you move this out of the class with this PR?

Reply:
I opted not to do it in this PR to limit the scope for the original RMv2 functionality adds.